Upgrade nippy to 2.13.0

Skiggz opened the pull request:
* The motivation here is actually to update the transient "com.taoensso/encore" dependency to a much newer version
* nippy 2.10.0 depends on a version of encore that dates back 2 years ago, 71 releases old x.x
* We had made this change before, and errors were being thrown, now we are making this change again and catching those
* When failing to deserialize old values, we will simply fail silently and assume the cache will be reloaded with a compatible serialized value
* Eventually (or never?) we can remove this but it shouldn't be detrimental in any case


@taubeklavs @skazhy @zarlen 

PS. I'm wondering if this should be a flag as well, that defaults to the try, but can avoid it if the client knows it's only got recently frozen items in cache (or cleared the cache).

Skiggz committed:
Upgrade nippy to 2.13.0

* The motivation here is actually to update the transient "com.taoensso/encore" dependency to a much newer version
* nippy 2.10.0 depends on a version of encore that dates back 2 years ago, 71 releases old x.x
* This treats deserialization like any other potential error caused during fetch, and throws if :throw? is true
* The theory here is, if upgrading from an old version, you'd get lots of cache misses until the cache slowly resets itself from old values

skazhy commented:
You can explicitly exclude nippy when adding full.cache as dependency & then adding nippy version which you need. This is a really high risk change, since there are lot of implicit dependencies on this.

FnTm commented on test/full/t_cache.clj:
```diff
@@ -3,12 +3,35 @@
+            [clojure.core.async :as async :refer [<!!]]
+            [taoensso.nippy :as nippy]))
+
+(def ^:private some-bytes
```
This should really be named better. Is there some specific meaning to these particular bytes? Or are they really just random bunch of bytes?

Skiggz replied:
They are random (well, they spell `FullContact!`) but it was for a test of a random input of bytes nippy wouldn't parse lol.

skazhy reviewed (changes requested):
As mentioned before - this change either needs to happen in the service using this, or bump version to `1.1.0`, so we can update cache with old nippy, if needed.

skazhy commented on src/full/cache.clj:
```diff
@@ -94,6 +94,20 @@
+    (catch Throwable e
+      (do
+        ;; Specifically do not log stack trace since that is slow and noisy and we know it was likely bad bytes
+        (log/debug "Failed to thaw value in cache for " k ", should be refreshed. Error: " (.getMessage e))
```
`full.core.log/debug` returns nil, so `do` block is redundant here.

Skiggz replied:
Oh yeah duh.

Skiggz commented:
Thanks for the feedback peeps.

I simplified this via karlis' suggestion. Now it's just a `2.0.0` version bump and I made deserialization a part of the normal fetch/throw flow that the interface is already built for (i.e. the `:throw?` option). I removed all the other mumbo jumbo so it's a pretty basic change. 

As for _risky_, I think the only risk here is 100% cache miss on day 1 that would quickly decline as the cache resets itself, assuming the cache is being used in normal fashion. If `:throw?` is being specified, then yes, error rates would spike until that flag was unset and/or the cache was cleared. For extra credit, I've added this note to the README.

Skiggz commented:
@skazhy bump in case you didn't see last comment/changes.

skazhy commented:
Rebased in master as `630065653b8`.
